packages/nvidia_nat_rag/src/nat/plugins/rag/rag_functions.py

```
@@ -0,0 +1,213 @@
+                        text = delta.get("content")
+                        if not text:
+                            message = choices[0].get("message", {})
+                            text = message.get("content", "")
+                        if text:
+                            full_response.append(text)
+                except json.JSONDecodeError:
+                    continue
+
+            return "".join(full_response) if full_response else "No response generated."
+
+        except Exception as e:
```

Ruff (0.14.11)

Do not catch blind exception: `Exception`

(BLE001)